src/dbnode/persist/fs/retriever.go

```diff
@@ -800,6 +800,17 @@ func (r *blockRetriever) Close() error {
 	return r.seekerMgr.Close()
 }
 
+func (r *blockRetriever) CloseShard(shard uint32) error {
+	r.Lock()
+	if r.status == blockRetrieverClosed {
```

Comment:
Nit: for such simple locked section, I would just assign `r.status` to a local var under a lock, and then do the `if` check outside of the lock. Having `Unlock` in a single place would make it slightly easier to read lock boundaries IMHO.

Reply:
In general I agree, but this would make linter unhappy with ` variable 'status' is only used in the if-statement`.